src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -865,9 +866,13 @@ protected override void DecodeWellKnownAttributeImpl(ref DecodeWellKnownAttribut
             }
             else if (attribute.IsTargetAttribute(AttributeDescription.UnscopedRefAttribute))
             {
-                if (!this.IsValidUnscopedRefAttributeTarget())
+                bool validTarget = this.IsValidUnscopedRefAttributeTarget();
+                if (!validTarget || !this.UseUpdatedEscapeRules)
```

Comment:
nit: In my opinion, it would be less "error prone" (no pun intended) to simply report the warning for this attribute in presence of old rules, regardless of value of `validTarget`. Then there is less risk of a small adjustment to this branching code, causing a warning to be reported instead of an error.